app/client/components/GridView.js

```diff
@@ -880,6 +881,15 @@ GridView.prototype.insertColumn = async function(colId = null, options = {}) {
   return newColInfo;
 };
 
+GridView.prototype.makeHeadersFromRow = async function(selection) {
+  for(const col of this.viewSection.columns.peek()) {
+    const colId = col.colId.peek();
+    const newColLabel = this.tableModel.tableData.getValue(selection.rowIds[0], colId);
+    const action = ['ModifyColumn', colId, {"label": newColLabel}];
+    this.tableModel.sendTableAction(action);
+  }
```

Comment:
There is a `sendTableActions` (with an `s` at the end) method that allows to send a single HTTP request for all the actions. I am not entirely sure by reading the code, but I guess this would also allow applying these actions in a single transaction (thus if an action fails, the other actions won't be applied):

```suggestion
  const actions = this.viewSection.columns.peek().map(col => {
    const colId = col.colId.peek();
    const newColLabel = this.tableModel.tableData.getValue(selection.rowIds[0], colId);
    return ['ModifyColumn', colId, {"label": newColLabel}];
  });
  this.tableModel.sendTableActions(actions);
```

Reply:
I'd recommend passing a description as the second argument too. Can be something short like `"Use as table headers"`.

Reply:
@georgegevoian If you see my last changes it's was I've done, is it good for you ? 
`this.tableModel.sendTableActions(actions, "Use as table headers");`